joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstForPrimitivesCreator.scala

```diff
@@ -0,0 +1,27 @@
+package io.joern.csharpsrc2cpg.astcreation
+
+import io.joern.csharpsrc2cpg.parser.DotNetJsonAst.*
+import io.joern.csharpsrc2cpg.parser.{DotNetNodeInfo, ParserKeys}
+import io.joern.x2cpg.{Ast, ValidationMode}
+import io.shiftleft.codepropertygraph.generated.nodes.NewCall
+import io.shiftleft.codepropertygraph.generated.{DispatchTypes, Operators}
+
+trait AstForPrimitivesCreator(implicit withSchemaValidation: ValidationMode) {
+  this: AstCreator =>
+  protected def astForIdentifier(ident: DotNetNodeInfo, typeFullName: String = ""): Ast = {
+    val identifierName = nameFromNode(ident)
+    if identifierName != "_" then {
+      val variableOption = scope.lookupVariable(identifierName)
+      variableOption match
+        case Some((variable, variableTypeName)) => {
+          val node = identifierNode(ident, identifierName, ident.code, variableTypeName)
+          Ast(node).withRefEdge(node, variable)
+        }
+        case _ =>
+          // TODO: Create ref edges to global identifiers
+          Ast(identifierNode(ident, identifierName, ident.code, typeFullName))
```

Comment:
I guess if we decide to remove fake methods then we won't need this, as this is related to fields